src/meltano/core/job/job.py

```diff
@@ -100,16 +101,18 @@ class Job(SystemModel):  # noqa: WPS214
 
     __tablename__ = "runs"
 
-    id: Mapped[int] = Column(types.Integer, primary_key=True)
-    job_name: Mapped[str] = Column(types.String)
-    run_id: Mapped[str] = Column(GUID, nullable=False, default=uuid.uuid4)
-    _state: Mapped[str] = Column(name="state", type_=types.String)
-    started_at: Mapped[datetime] = Column(types.DateTime)
-    last_heartbeat_at: Mapped[datetime] = Column(types.DateTime)
-    ended_at: Mapped[datetime] = Column(types.DateTime)
-    payload: Mapped[dict] = Column(MutableDict.as_mutable(JSONEncodedDict))
-    payload_flags: Mapped[Payload] = Column(IntFlag, default=0)
-    trigger: Mapped[str] = Column(types.String, default=current_trigger)
+    id: Mapped[IntPK]
+    job_name: Mapped[t.Optional[str]]  # noqa: UP007
+    run_id: Mapped[GUIDType]
+    _state: Mapped[t.Optional[str]] = mapped_column(name="state")  # noqa: UP007
+    started_at: Mapped[t.Optional[datetime]]  # noqa: UP007
+    last_heartbeat_at: Mapped[t.Optional[datetime]]  # noqa: UP007
+    ended_at: Mapped[t.Optional[datetime]]  # noqa: UP007
+    payload: Mapped[dict] = mapped_column(MutableDict.as_mutable(JSONEncodedDict))
+    payload_flags: Mapped[Payload] = mapped_column(IntFlag, default=0)
+    trigger: Mapped[t.Optional[str]] = mapped_column(  # noqa: UP007
+        default=current_trigger,
+    )
```

Comment:
Let me know if these are too noisy here and I can move the typing changes to a new PR